cli/cmd/utils_internal_test.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/spf13/viper"
+)
+
+type kubeconfigSources struct {
+	flag       string
+	env        string
+	configFile string
+}
+
+func prepareKubeconfigSource(t *testing.T, k *kubeconfigSources) {
+	// Ensure viper flag is NOT empty.
+	viper.Set(kubeconfigFlag, k.flag)
+
+	if k.env == "" {
+		// Ensure KUBECONFIG is not set.
+		if err := os.Unsetenv(kubeconfigEnvVariable); err != nil {
+			t.Fatalf("unsetting %q environment variable: %v", kubeconfigEnvVariable, err)
+		}
+	}
+
+	if k.env != "" {
+		// Ensure KUBECONFIG IS set.
+		if err := os.Setenv(kubeconfigEnvVariable, k.env); err != nil {
+			t.Fatalf("setting %q environment variable: %v", kubeconfigEnvVariable, err)
+		}
+	}
+
+	// Ensure there is no lokocfg configuration in working directory.
+	tmpDir, err := ioutil.TempDir("", "lokoctl-tests-")
+	if err != nil {
+		t.Fatalf("creating tmp dir: %v", err)
+	}
+
+	t.Cleanup(func() {
+		if err := os.RemoveAll(tmpDir); err != nil {
+			t.Logf("removing temp dir %q: %v", tmpDir, err)
+		}
+	})
+
+	if err := os.Chdir(tmpDir); err != nil {
+		t.Fatalf("changing working directory to %q: %v", tmpDir, err)
+	}
+
+	if k.configFile != "" {
+		path := filepath.Join(tmpDir, "cluster.lokocfg")
+		if err := ioutil.WriteFile(path, []byte(k.configFile), 0600); err != nil {
+			t.Fatalf("writing file %q: %v", path, err)
+		}
+	}
+}
+
+func TestGetKubeconfigFlag(t *testing.T) {
+	expectedPath := "/foo"
+
+	k := &kubeconfigSources{
+		flag: expectedPath,
```

Comment:
I'd add configFile, and env here, to ensure that file takes precedence over everything else.